### PR TITLE
Clean up differential expression workflow

### LINKED ISSUE #144: Decide how to proceed with the differential expression subworkflow
### Description of feature

The way it is currently implemented is not state of the art. Especially the way the linear quantification is currently implemented is less than optimal. The results of the `QUANTIFICATION` subworkflow should be used here.

```[tasklist]
### Tasks
- [ ] Remove linear quantification in `DIFFERENTIAL_EXPRESSION`
- [ ] Replace DESeq2 with Fishpond/Swish
- [ ] Update `circtest` implementation
```

### PR DESCRIPTION
**Pull request by nictru**
- Implements the changes proposed in [this slack message](https://nfcore.slack.com/archives/C01MKA377LM/p1718102338402999)
- Moves most scripts from the `bin` directory to the corresponding modules using the `template` syntax
- Deletes some artifact files
- Simplifies the `circtest` implementation

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,7 +22,7 @@
 
 ## Introduction
 
-**nf-core/circrna** is a bioinformatics best-practice analysis pipeline for Quantification, miRNA target prediction and differential expression analysis of circular RNAs.
+**nf-core/circrna** is a bioinformatics best-practice analysis pipeline for BSJ detection, quantification, annotation and miRNA target prediction of circular RNAs.
```
Comment (mahesh-panchal) [inline README.md]: Perhaps put what the acronym BSJ stands for parentheses.

File: docs/usage.md [hunk 2]
```diff
@@ -167,9 +149,9 @@ A valid example of a `phenotype.csv` file (matching the TCGA example input CSV f
 
 `nf-core/circrna` provides 3 analysis modules to the user:
 
-1. circRNA quantification & annotation.
-2. miRNA target prediction.
-3. Differential circRNA expression analysis.
+1. BSJ detection
+2. Common quantification of circular and linear transcriptome
```
Comment (mahesh-panchal) [inline docs/usage.md]: What does "common quantification" mean?

File: modules/local/annotation/main.nf [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+        'biocontainers/pandas:1.5.2' }"
+
+    input:
+    tuple val(meta), path(intersection)
+    val(exon_boundary)
+
+    output:
+    tuple val(meta), path("${prefix}.bed"), emit: bed
+    tuple val(meta), path("${prefix}.gtf"), emit: gtf
+
```
Comment (mahesh-panchal) [inline modules/local/annotation/main.nf:17]:
```suggestion
```

File: modules/local/circtest/circtest/main.nf [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+    input:
+    tuple val(meta) , path(circ_counts)
+    tuple val(meta2), path(gene_counts)
+    tuple val(meta3), path(phenotype)
+
+    output:
+    tuple val(meta), path("${prefix}_summary.txt"), emit: summary
+    tuple val(meta), path("*.pdf")                , emit: plots
+
```
Comment (mahesh-panchal) [inline modules/local/circtest/circtest/main.nf]:
```suggestion
```

**Review by mahesh-panchal — APPROVED** [review]
LGTM generally.

File: README.md (hunk 1 above)
Comment (nictru) [reply README.md]: Happens when you are too used to the acronyms - will clarify

File: docs/usage.md (hunk 2 above)
Comment (nictru) [reply docs/usage.md]: Do you thing "Joint quantification" is clear enough? 
It should basically mean that there is a run of kallisto (actually a wrapper called psirc-quant), which takes a single fasta file containing both the known linear and detected circular transcriptome.

Comment (mahesh-panchal) [reply docs/usage.md]: That sounds good.

**Pull request merged**